website/project/model.py

```diff
@@ -1587,6 +1594,10 @@ def save(self, *args, **kwargs):
         if settings.PIWIK_HOST and update_piwik:
             piwik_tasks.update_node(self._id, saved_fields)
 
+        # Ensure Mailing list subscription is set up
+        if first_save and self.mailing_enabled:
+            auth_signals.node_created.send(self)
```

Comment:
The (currently only) function signaled by this does a check to ensure that `node.mailing_enabled`, I put this here as a slight optimization. Pending your response to [this comment](https://github.com/CenterForOpenScience/osf.io/pull/5033#discussion_r53787074), I'll change this.